specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AMLComputeConfiguration": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the AMLCompute cluster.",
+          "type": "string",
+          "example": "my8nodeCluster"
+        },
+        "vmSize": {
+          "description": "VM size of the Cluster to be created. Allowed values are Azure vm sizes.\r\nThe list of vm sizes is available in https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-sizes-specs",
+          "type": "string",
+          "example": "Standard_D2_v2"
+        },
+        "vmPriority": {
+          "description": "VM priority of the Cluster to be created. Allowed values are dedicated and lowpriority.",
+          "enum": [
+            "Dedicated",
+            "Lowpriority"
+          ],
+          "type": "string",
+          "example": "dedicated",
+          "x-ms-enum": {
+            "name": "VmPriority",
+            "modelAsString": true
+          }
+        },
+        "retainCluster": {
+          "description": "Setting to true will prevent the cluster from being deleted upon completion of the run.",
+          "type": "boolean",
+          "example": "False"
+        },
+        "clusterMaxNodeCount": {
+          "format": "int32",
+          "description": "The maximum number of nodes that the cluster can scale up to.",
+          "type": "integer",
+          "example": "10"
+        }
+      }
+    },
+    "TensorflowConfiguration": {
+      "type": "object",
+      "properties": {
+        "workerCount": {
+          "format": "int32",
+          "description": "The number of workers.",
+          "type": "integer",
+          "example": "2"
+        },
+        "parameterServerCount": {
+          "format": "int32",
+          "description": "Number of parameter servers.",
+          "type": "integer",
+          "example": "1"
+        }
+      }
+    },
+    "MpiConfiguration": {
+      "type": "object",
+      "properties": {
+        "processCountPerNode": {
+          "format": "int32",
+          "description": "Number of processes per node.",
+          "type": "integer",
+          "example": "2"
+        }
+      }
+    },
+    "HdiConfiguration": {
+      "type": "object",
+      "properties": {
+        "yarnDeployMode": {
+          "enum": [
+            "None",
+            "Client",
+            "Cluster"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "YarnDeployMode",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ContainerInstanceConfiguration": {
+      "type": "object",
+      "properties": {
+        "region": {
+          "description": "Defaults to the region of the workspace.",
+          "type": "string",
+          "example": "eastus2"
+        },
+        "cpuCores": {
+          "format": "double",
```

Comment:
Is this really a `double` and not an integer?

Reply:
It really is!

We forward this value to ACI and they take a double for the CPU core limit:
https://docs.microsoft.com/en-us/rest/api/container-instances/containergroups/createorupdate

Reply:
done